rskj-core/src/main/java/co/rsk/peg/federation/FederationContext.java

```diff
@@ -23,21 +25,43 @@ public Optional<Federation> getRetiringFederation() {
         return Optional.ofNullable(retiringFederation);
     }
 
-    public void setRetiringFederation(Federation retiringFederation) {
-        this.retiringFederation = retiringFederation;
-    }
-
     public Optional<Script> getLastRetiredFederationP2SHScript() {
         return Optional.ofNullable(lastRetiredFederationP2SHScript);
     }
 
-    public void setLastRetiredFederationP2SHScript(Script lastRetiredFederationP2SHScript) {
-        this.lastRetiredFederationP2SHScript = lastRetiredFederationP2SHScript;
-    }
-
     public List<Federation> getLiveFederations() {
         return Stream.of(activeFederation, retiringFederation)
             .filter(Objects::nonNull)
             .toList();
     }
+
+    public static FederationContextBuilder builder() {
+        return new FederationContextBuilder();
+    }
+
+    public static class FederationContextBuilder {
+        private Federation activeFederation;
+        private Federation retiringFederation;
+        private Script lastRetiredFederationP2SHScript;
+
+        public FederationContextBuilder setActiveFederation(Federation activeFederation) {
```

Comment:
```suggestion
        public FederationContextBuilder withActiveFederation(Federation activeFederation) {
```
Just to keep it consistent witht he other builders we have. Same applies for the remaining attributes

Reply:
updated!